test/libsolidity/syntaxTests/tupleAssignments/warn_multiple_storage_storage_copies.sol

```diff
@@ -4,6 +4,11 @@ contract C {
 	function f() public {
 		(x, y) = (y, x);
 	}
+	function g() public {
+		uint z;
+		((x, y), z) = ((y, x), 0);
+	}
 }
 // ----
 // Warning 7238: (79-94): This assignment performs two copies to storage. Since storage copies do not first copy to a temporary location, one of them might be overwritten before the second is executed and thus may have unexpected effects. It is safer to perform the copies separately or assign to storage pointers first.
+// Warning 7238: (134-159): This assignment performs two copies to storage. Since storage copies do not first copy to a temporary location, one of them might be overwritten before the second is executed and thus may have unexpected effects. It is safer to perform the copies separately or assign to storage pointers first.
```

Comment:
Simultaneous reads to the same locations are what the warning is for, but it has false positives, so it will also warn about unrelated copies, i.e. for
```
uint[] x;
uint[] y;
uint[] z;
uint[] w;
function f(bool c) public {
   uint[] storage r = c ? y : z;
   uint[] storage r2 = c ? x : w;
   (x,y) = (r,r2);
}
```
it's impossible to tell at compile time if the case is fine or not, so it also warns for
```
uint[] x;
uint[] y;
uint[] z;
uint[] w;
function f() public {
   uint[] storage r = z;
   uint[] storage r2 = w;
   (x,y) = (r,r2);
}
```
even though a deep analysis could realize that it's fine.